### ISSUE TITLE
Concurrent access results in inconsistent results

### ISSUE THREAD

**Issue by Abdelgha-4** [ASMfreaK/aiotinydb#15]
**How to reproduce:**
script1.py:
```
import asyncio
from aiotinydb import AIOTinyDB

async def test():
    async with AIOTinyDB('test.json') as db:
        await asyncio.sleep(60)  # slow operation in script1
        db.insert(dict(counter=1))

loop = asyncio.new_event_loop()
loop.run_until_complete(test())
loop.close()
```
script2.py:
```
import asyncio
from aiotinydb import AIOTinyDB

async def test():
    async with AIOTinyDB('test.json') as db:
        # insert directly, no waiting
        db.insert(dict(counter=2))

loop = asyncio.new_event_loop()
loop.run_until_complete(test())
loop.close()
```

Run script1, then immediately run the script2 (before the 60seconds waiting time).

**Expected Output:**
the script2 should be the first inserting a value, then after the sleep statement the script1 should then insert the second value
`{"_default": {"1": {"counter": 2}, "2": {"counter": 1}}}`

**Actual Output:**
Apparently the script1 overrides the value inserted by the script2, and the value `{"counter": 2}` is lost !
`{"_default": {"1": {"counter": 1}}}`

**Environement:**

```
python                : 3.12.4.final.0
python-bits           : 64
OS                    : Windows
OS-release            : 10
Version               : 10.0.19045
machine               : AMD64
processor             : Intel64 Family 6 Model 142 Stepping 9, GenuineIntel
byteorder             : little
setuptools            : 70.3.0
pip                   : 24.0
aiotinydb             : 2.0.0
```

**Comment by d-k-bo** [comment]
I guess you are on Windows? Accessing the database from multiple processes is only supported on unix-like systems where file locking is used to prevent such race conditions.

**Comment by asmfreak** [comment]
@d-k-bo We may want to implement file locking on windows too. I don't own a windows machine to test, but there is a [windows alternative to flock](https://docs.python.org/3/library/msvcrt.html#msvcrt.locking). It does not have a non-blocking mode, so we must run it in an executor. 

We can also look into an [alternative library](https://py-filelock.readthedocs.io/en/latest/), but it also requires running in an executor. It would also require using a separate file for synchronization.  

The latter approach may be better, because a separate lockfile seems to be a better way.

**Comment by Abdelgha-4** [comment]
@d-k-bo Yes I confirm I'm on windows, I've just added my environnement info to the issue.

**Comment by d-k-bo** [comment]
I don't use windows either, but I have a virtual machine image somewhere that I could use for final testing.

I think py-filelock is the way to go. It seems like they recently added support for asyncio, although the documentation isn't yet up to date. I'll look into it, however, I'm still a little busy at the moment with other projects.
